source/draw/gpu/opengl/DisplayContext.ooc

```diff
@@ -0,0 +1,16 @@
+/* This file is part of magic-sdk, an sdk for the open source programming language magic.
+ *
+ * Copyright (C) 2016 magic-lang
+ *
+ * This software may be modified and distributed under the terms
+ * of the MIT license.  See the LICENSE file for details.
+ */
+
+use base
+import backend/[GLFence, GLContext]
+
+DisplayContext: abstract class {
+	getDisplay: abstract func -> Pointer
+	printExtensions: abstract func
+	swapBuffers: abstract func
```

Comment:
I plan to place native fence constructors here to hide more dependency to EGL  but the type safety is making it hard to create the constructors without returning generic types that would break backward compatibility and require lots of testing.